azurelinuxagent/common/utils/restutil.py

```diff
@@ -150,6 +150,22 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         self._conn.close()
 
 
+class HTTPResponseMaterialized(object):
+    def __init__(self, resp):
+        self.status = resp.status
+        self.reason = resp.reason
+        self._content = resp.read()
```

Comment:
Definitely not what I imagined. This context object actually results in the response object being destroyed immediately (after capturing a few, possibly expensive, attributes). Nothing "active" is done when __exit__ is called, because there's nothing to do.

My expectation was that this object would literally hold on to the response object for its entire lifetime, with __enter__() returning (a reference to) the wrapped request object which would be explicitly closed in the __exit__() method.